test/class_test.rb

```diff
@@ -1,80 +1,88 @@
 require "test_helper"
 
-[::Object, ::Toy].each do |ns|
-  describe "Class-like behaviour in the #{ns} namespace" do
-    [[ns::Class, ns::Object]].each do |_Class, _Object|
-      describe _Class do
+module ClassBehaviourTests
+  def self.included(_)
```

Comment:
I'm used to seeing `ActiveSupport::Concern#included` used to define behaviour when a Module is included. Are the two (`ActiveSupport::Concern#included` and `Module#included`) essentially the same in practice?

Reply:
Yeah, `ActiveSupport::Concern` gives you a nice DSL for it (`included do … end`) rather than making you `def` a method, and it automatically evaluates the block in the context of the `base` argument rather than the module being included, but it’s doing pretty much the same thing. (One small difference is that `ActiveSupport::Concern#included` actually [runs the block](https://github.com/rails/rails/blob/v7.0.1/activesupport/lib/active_support/concern.rb#L136) during [`Module#append_features`](https://rubydoc.info/stdlib/core/Module:append_features) rather than `Module#included`, even though [the `Module` docs](https://rubydoc.info/stdlib/core/Module:included) say to prefer the latter; I’m sure Rails has its reasons.)

The `ActiveSupport::Concern` [docs](https://rubydoc.info/gems/activesupport/ActiveSupport/Concern) are surprisingly helpful in explaining why its DSL methods exist and what their corresponding bare-Ruby counterparts looks like.

Reply:
>  it automatically evaluates the block in the context of the base argument rather than the module being included

Ohh, neat! So `self` in an `ActiveSupport::Concern#included` block would refer to the object _including_ the module, whereas `self` in a `Module#included` method def refers to the _module_ defining the method, and takes an argument which is the object including the module.

Those docs were super helpful! Thanks for the context ❤️